src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -117,32 +116,28 @@ public boolean checkInWareHouseForFood(final BuildingHome buildingEntry, final b
                 if (addToList)
                 {
                     buildingEntry.setOnGoingDelivery(true);
-                    list.add(buildingEntry);
+                    taskQueue.add(buildingEntry);
                 }
                 return true;
             }
 
-            if (list.contains(buildingEntry))
+            if (taskQueue.contains(buildingEntry))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)